### DISCUSSION TITLE
Example using Keycloak in the Kubernetes RabbitMQ Cluster Operator

### DISCUSSION THREAD

**Discussion by psavva** [rabbitmq/rabbitmq-server#11100] [Other] [answered]
### Is your feature request related to a problem? Please describe.

No clear example of how to configure RabbitMQ with Keycloak using the Kubernetes Cluster Operator

### Describe the solution you'd like

An working example using the Kubernetes operator

**Comment by michaelklishin** [comment, marked as answer]
There are extensive doc guides for

 * The [OAuth 2 plugin](https://www.rabbitmq.com/docs/oauth2)
 * OAuth 2 [examples for five different IDPs](https://www.rabbitmq.com/docs/oauth2-examples), including Keycloak
 * Pretty extensive [RabbitMQ Kubernetes Operator](https://www.rabbitmq.com/kubernetes/operator/operator-overview) example

I'm afraid our small team cannot produce five more examples with different IDPs on Kubernetes, and I am not sure an example Keycloak (or any other IDP) Kubernetes installation would be a good fit for RabbitMQ docs. How exactly the IDP is provisioned in your Kubernetes cluster is up to you to decide. 

Finally, RabbitMQ is open source software you @psavva very likely get for free, include [the open source documentation](https://github.com/rabbitmq/rabbitmq-website), so anyone can contribute an example or simply publish it on your own blog and promote it on community Discord, Discussions, the mailing list and so on.

**Reply by psavva** [reply]
I will attempt to provide a working examples for kubernetes and Keycloak and submit a PR
